manifests/charts/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -250,5 +256,7 @@ spec:
         configMap:
           name: pilot-jwks-extra-cacerts{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
   {{- end }}
-
+      {{- if .Values.pilot.volumes }}}
```

Comment:
https://github.com/istio/istio/blob/c8b482f5bdfb6eef2970d85928feac7f6375d9c8/manifests/charts/istio-control/istio-discovery/templates/deployment.yaml#L235-L239
It will mount here if you have the cacerts.

Reply:
Aware of this. Added some additional reasoning for the PR here: https://github.com/istio/istio/pull/45323#issuecomment-1583355372